src/benchmark/java/com/commercetools/sync/benchmark/CartDiscountSyncBenchmark.java

```diff
@@ -0,0 +1,281 @@
+        final long totalTime = System.currentTimeMillis() - beforeSyncTime;
+
+        // assert on threshold (based on history of benchmarks; highest was ~13 seconds)
+        final int threshold = 26000; // double of the highest benchmark
+        assertThat(totalTime).withFailMessage(format(THRESHOLD_EXCEEDED_ERROR, totalTime, threshold))
+                             .isLessThan(threshold);
+
+        // Assert actual state of CTP project (total number of existing cart discounts)
+        final CompletableFuture<Integer> totalNumberOfCartDiscounts =
+            CTP_TARGET_CLIENT.execute(CartDiscountQuery.of())
+                             .thenApply(PagedQueryResult::getTotal)
+                             .thenApply(Long::intValue)
+                             .toCompletableFuture();
+
+        executeBlocking(totalNumberOfCartDiscounts);
+        assertThat(totalNumberOfCartDiscounts).isCompletedWithValue(NUMBER_OF_RESOURCE_UNDER_TEST);
+
+        assertThat(syncStatistics).hasValues(NUMBER_OF_RESOURCE_UNDER_TEST, NUMBER_OF_RESOURCE_UNDER_TEST, 0, 0);
+        assertThat(errorCallBackExceptions).isEmpty();
+        assertThat(errorCallBackMessages).isEmpty();
+        assertThat(warningCallBackMessages).isEmpty();
+
+        saveNewResult(CART_DISCOUNT_SYNC, CREATES_ONLY, totalTime);
+    }
+
+    @Test
+    void sync_ExistingCartDiscounts_ShouldUpdateCartDiscounts() throws IOException {
+        // preparation
+        final List<CartDiscountDraft> cartDiscountDrafts = buildCartDiscountDrafts(NUMBER_OF_RESOURCE_UNDER_TEST);
+
+        // Create cart discount drafts to target project
+        // with different stacking mode (STOP_AFTER_THIS_DISCOUNT)
+        CompletableFuture.allOf(
+            cartDiscountDrafts.stream()
+                              .map(CartDiscountDraftBuilder::of)
+                              .map(builder -> builder.stackingMode(StackingMode.STOP_AFTER_THIS_DISCOUNT))
+                              .map(CartDiscountDraftBuilder::build)
+                              .map(draft -> CTP_TARGET_CLIENT.execute(CartDiscountCreateCommand.of(draft)))
+                              .map(CompletionStage::toCompletableFuture)
+                              .toArray(CompletableFuture[]::new))
+                         .join();
+
+        final CartDiscountSync cartDiscountSync = new CartDiscountSync(cartDiscountSyncOptions);
+
+
+        // benchmark
+        final long beforeSyncTime = System.currentTimeMillis();
+        // Update cart discount drafts to target project with different stacking mode (STACKING)
+        final CartDiscountSyncStatistics syncStatistics = executeBlocking(cartDiscountSync.sync(cartDiscountDrafts));
+        final long totalTime = System.currentTimeMillis() - beforeSyncTime;
+
+        // assert on threshold (based on history of benchmarks; highest was ~13 seconds)
+        final int threshold = 26000; // double of the highest benchmark
```

Comment:
I guess this is not true for this benchmark yet ;)

Reply:
what is your suggestion and how did you calculate these things before benchmarks?

Reply:
you can run it locally on your machine. Get the time it took to finish for each benchmark and use its double as the threshold.

Reply:
50527df